client/keystore/src/lib.rs

```diff
@@ -97,25 +98,22 @@ impl Store {
 		}))
 	}
 
-	/// Get the public/private key pair for the given public key and key type.
-	fn get_additional_pair<Pair: PairT>(
+	/// Get the key phrase for the given public key and key type from the in-memory store.
+	fn get_additional_pair(
 		&self,
-		public: &Pair::Public,
+		public: &[u8],
 		key_type: KeyTypeId,
-	) -> Result<Option<Pair>> {
-		let key = (key_type, public.to_raw_vec());
-		self.additional
-			.get(&key)
-			.map(|bytes| Pair::from_seed_slice(bytes).map_err(|_| Error::InvalidSeed))
-			.transpose()
+	) -> Option<&String> {
+		let key = (key_type, public.to_vec());
+		self.additional.get(&key)
 	}
 
 	/// Insert the given public/private key pair with the given key type.
 	///
 	/// Does not place it into the file system store.
-	fn insert_ephemeral_pair<Pair: PairT>(&mut self, pair: &Pair, key_type: KeyTypeId) {
+	fn insert_ephemeral_pair<Pair: PairT>(&mut self, pair: &Pair, seed: &str, key_type: KeyTypeId) {
 		let key = (key_type, pair.public().to_raw_vec());
-		self.additional.insert(key, pair.to_raw_vec());
+		self.additional.insert(key, seed.into());
```

Comment:
The docs for `additional` say it should be `raw public key`, but here it seems we just put `seed` in.

Also we can just put bulshit, cause we don't check that the `pair` is derived from that `seed`, do we?

Reply:
The docs say: `(KeyTypeId, Raw public key)` -> `Key phrase/seed`
                                      Key                            ->       Value

(the key is build above in exactly this way.